# NB 9: - Grading the professor, Pt. 3

*Modified from Data Science in a box*

In [5]:
# This code will load the R packages we will use
install.packages(c("csucistats", "openintro"),
                 repos = c("https://inqs909.r-universe.dev", "https://cloud.r-project.org"))
library(csucistats)
library(tidyverse)
library(openintro)


# Uncomment and run for categorical plots
# csucistats::install_plots()
# library(ggtricks)
# library(ggmosaic)
# library(waffle)

# Uncomment and run for themes
# csucistats::install_themes()
# library(ThemePark)
# library(ggthemes)


Installing packages into ‘/home/inqs/R/x86_64-redhat-linux-gnu-library/4.4’
(as ‘lib’ is unspecified)


trying URL 'https://inqs909.r-universe.dev/src/contrib/csucistats_0.4.3.tar.gz'
Content type 'application/gzip' length 2183582 bytes (2.1 MB)
downloaded 2.1 MB

trying URL 'https://cloud.r-project.org/src/contrib/openintro_2.5.0.tar.gz'
Content type 'application/x-gzip' length 5381706 bytes (5.1 MB)
downloaded 5.1 MB

* installing *source* package ‘csucistats’ ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices


  converting help for package ‘csucistats’


    finding HTML links ... done


    ar2                                     html  
    b                                       html  
    bottom                                  html  
    cat_stats                               html  
    csucistats-package                      html  
    descriptive                             html  
    f                                       html  
    install_m201                            html  
    install_plots                           html  
    install_themes                          html  
    iterate                                 html  
    middle                                  html  
    num_by_cat_stats                        html  
    num_stats                               html  
    props                                   html  
    r2                                      html  
    resample                                html  
    resid_df                                html  
    se_b                                    html  
    shuffle                    

*** copying figures
** building package indices
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (csucistats)
* installing *source* package ‘openintro’ ...
** package ‘openintro’ successfully unpacked and MD5 sums checked
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices


  converting help for package ‘openintro’


    finding HTML links ... done


    ArrowLines                              html  
    AxisInDollars                           html  
    AxisInPercent                           html  
    BG                                      html  
    Braces                                  html  
    CCP                                     html  
    COL                                     html  
    CT2DF                                   html  
    ChiSquareTail                           html  
    IMSCOL                                  html  
    LAhomes                                 html  
    MosaicPlot                              html  
    PlotWLine                               html  
    absenteeism                             html  
    acs12                                   html  
    age_at_mar                              html  
    ames                                    html  
    ami_occurrences                         html  
    antibiotics                             html  
    arbuthnot                  

*** copying figures
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (openintro)

The downloaded source packages are in
	‘/tmp/RtmpEGr0lm/downloaded_packages’


Loading required package: airports
Loading required package: cherryblossom
Loading required package: usdata

Attaching package: ‘openintro’

The following object is masked from ‘package:mosaic’:

    dotPlot

The following object is masked from ‘package:dslabs’:

    murders

The following objects are masked from ‘package:lattice’:

    ethanol, lsegments



## Introduction

Many college courses conclude by giving students the opportunity to evaluate the course and the instructor anonymously.
However, the use of these student evaluations as an indicator of course quality and teaching effectiveness is often criticized because these measures may reflect the influence of non-teaching related characteristics, such as the physical appearance of the instructor.
The article titled, "Beauty in the classroom: instructors' pulchritude and putative pedagogical productivity" (Hamermesh and Parker, 2005) found that instructors who are viewed to be better looking receive higher instructional ratings.
(Daniel S. Hamermesh, Amy Parker, Beauty in the classroom: instructors pulchritude and putative pedagogical productivity, Economics of Education Review, Volume 24, Issue 4, August 2005, Pages 369-376, ISSN 0272-7757, 10.1016/j.econedurev.2004.07.013. <http://www.sciencedirect.com/science/article/pii/S0272775704001165>.)





## Data

In this notebook you will analyze the data from this study in order to learn what goes into a positive professor evaluation.

The data were gathered from end of semester student evaluations for a large sample of professors from the University of Texas at Austin.
In addition, six students rated the professors' physical appearance.
(This is a slightly modified version of the original data set that was released as part of the replication data for Data Analysis Using Regression and Multilevel/Hierarchical Models (Gelman and Hill, 2007).) The result is a data frame where each row contains a different course and columns represent variables about the courses and professors.

The data can be found in the **openintro** package, and it's called `evals`.
Since the dataset is distributed with the package, we don't need to load it separately; it becomes available to us when we load the package.
You can find out more about the dataset by inspecting its documentation, which you can access by running `?evals` in the Console or using the Help menu in RStudio to search for `evals`.
You can also find this information [here](https://www.openintro.org/data/index.php?data=evals).




## 1.0 - Linear regression with a numerical predictor

1.1 - Fit a linear model predict average professor evaluation `score` (outcome) by average beauty rating (`bty_avg`).
    Based on the regression output, write the linear model.



1.2 - Interpret the slope of the linear model in context of the data.



1.3 - Write the null and alternative hypothesis to determine if `bty_avg` and `score` are associated or random.

1.4 - What is the statistic of interest?

1.5 Use the p-value approach to test the null and alternative hypothesis.

1.6 Interpret and draw conclusions from the results that you found.


## 2.0 Linear regression with a categorical predictor



2.1 -  Fit a new linear model to predict average professor evaluation `score` (outcome) based on `gender` of the professor.
    Based on the regression output, write the linear model.



2.2 - Write the null and alternative hypothesis seeing th scores are different by gender.


2.3 - What Statistic would you use to test the null hypothesis?

2.4 - Use the confidence interval approach to test the null and alternative hypothesis.

2.5 - Interpret and draw conclusions from the results that you found.

## 3.0 Multiple Linear Regression

3.1 -  Fit a new linear model to predict average professor evaluation `score` (outcome) based on `gender` of the professor, `rank` of the professor, and `bty_avg`. Based on the regression output, write the linear model.

3.2 - Write the null and alternative hypothesis seeing th scores are affected by beauty average, adjusting for rank and gender.


3.3 Conduct the hypothesis test written in 3.2.

3.4 Interpret the results of the Hypothesis Test.

## 4.0 Submit Notebook

Submit the notebook to Canvas.